src/lib.rs

```diff
@@ -1,4 +1,103 @@
 mod client;
 mod util;
+use futures::{future, Async, Poll};
+use http::{Request, Response};
+use hyper::{
+    body::Payload,
+    client::connect::Connect as HyperConnect,
+    client::{HttpConnector, ResponseFuture},
+    Body, Client as HyperClient,
+};
+use tower_retry::Policy;
+use tower_service::Service;
 
 pub use self::client::{Connect, Connection};
+
+pub struct Client<C = HttpConnector, B = Body> {
+    inner: HyperClient<C, B>,
+}
+
+impl<C, B> Client<C, B> {
+    fn new(inner: HyperClient<C, B>) -> Self {
+        Self { inner }
+    }
+}
+
+impl<C, B> Service<Request<B>> for Client<C, B>
```

Comment:
only a minor thought but prob want to implement DirectService where all the `poll_*` just return ready. This will ease the transition from `Service` -> `DirectService` which should happen this week or next 🙏

Reply:
that's probably a good idea, will do.